### PR TITLE
Backend API for uploading resources

### PR DESCRIPTION
**Pull request by AdnanMuzavor**
## Issue that this pull request solves

 Closes:#736

## Proposed changes:
->Created Model for resources as specified
->Created API for posting Resources
->If important fields are not filled specific error is thrown
->If schema is not followed i.e string used instead of int, error is thrown
## Types of changes

_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update (Documentation content changed)
- [x] Other (please describe): Added an ERROR  constant " REQEMPTYFIELDS", It specifies that certain important fields cannot be left empty.

## Checklist

_Put an `x` in the boxes that apply_

- [x] My code follows the style guidelines of this project
- [x ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x ] I have made corresponding changes to the documentation
- [x ] My changes generate no new warnings
- [x ] My changes does not break the current system and it passes all the current test cases.

## Screenshots

Please attach the screenshots of the changes made in case of change in user interface
[](url)
## Other information

resources model screen shot: 
Screenshot (697)

addresource function:
Screenshot (698)

Screenshot (699)

index.js in resource folder:
Screenshot (700)
 
index.js in routes folder:
Screenshot (701)

constants.json (to show additional constant added):
Screenshot (702)

Any other information that is important to this pull request

Here is recording of API testing: https://www.recordjoy.com/view/39lxUuSciSeGUoM5AWuH
1)When all content is properly provided
2)When something required is missing example "name"
3)When wrong data types are used(string in place of integer)

### CODE REVIEW

File: backend/app/routes/resources/addResource.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+//Getting resource modal
+const Resource = require('../../models/resource');
+//Getting resource handler
+const { ErrorHandler } = require('../../../helpers/error');
+const constants = require('../../../constants');
+
+module.exports = async (req, res, next) => {
+  try {
+    //Destructuring fields which are must
+    const { name, email, url, trustLevel, expiryDate } = req.body;
```
Comment (Kajol-Kumari) [inline backend/app/routes/resources/addResource.js:10]: We use the validation Schema to add check for the required field. Please see https://github.com/HITK-TECH-Community/Community-Website/pull/695/files#diff-66eaadabdfb2a1e580497efec3551b7ff1a144afc8f0d2732ba5643b0032f3c3R1 this pr for reference

File: backend/app/routes/resources/addResource.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      const error = new ErrorHandler(constants.ERRORS.DATABASE, {
+        statusCode: 500,
+        message: 'Database Error',
+        user: req.body.email,
+        errStack: 'Unable to post resource',
+      });
+      return next(error);
+    }
+    //Else sending data stored as response
+    res.status(200).send(newResource);
```
Comment (Kajol-Kumari) [inline backend/app/routes/resources/addResource.js:34]: After successful addition, we want to send the email to all the admins. Please refer https://github.com/HITK-TECH-Community/Community-Website/pull/695/files#diff-570f5e87cc8f64b44733c6e3209357e3450f7fb88a038466acd1c43781150d28R30 for the reference

File: backend/constants/constants.json [hunk 3]
```diff
@@ -6,7 +6,9 @@
     "CRITICAL": "Critical Error",
     "WARNING": "Warning",
     "USER": "User ",
-    "UNEXPECTED": "Unexpected Error"
+    "UNEXPECTED": "Unexpected Error",
+    "REQEMPTYFIELDS":"Required Fields Not Filled"
```
Comment (Kajol-Kumari) [inline backend/constants/constants.json:10]: We won't need this variable after implementing the above commented approach. So, you can get rid of it.

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
@AdnanMuzavor please make the requested changes and also add the test for this api

File: backend/constants/constants.json [hunk 4]
```diff
@@ -7,6 +7,8 @@
     "WARNING": "Warning",
     "USER": "User ",
     "UNEXPECTED": "Unexpected Error"
+
```
Comment (Kajol-Kumari) [inline backend/constants/constants.json:10]: please revert the change of this file

File: backend/app/routes/resources/index.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+//Creating a router
```
Comment (Kajol-Kumari) [inline backend/app/routes/resources/index.js:1]: please remove the comments from this file as imports are understandable

File: backend/app/routes/resources/addResource.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
+const to = require('await-to-js').default;
+//Getting resource modal
```
Comment (Kajol-Kumari) [inline backend/app/routes/resources/addResource.js:2]: please remove comment from imports

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
@AdnanMuzavor please do the minor requested changes.

File: backend/app/routes/resources/addResource.js (hunk 6 above)
Comment (AdnanMuzavor) [reply backend/app/routes/resources/addResource.js:2]: Okay Ma'am.

**Review by Kajol-Kumari — APPROVED** [review]
Thanks for making the required changes @AdnanMuzavor 
LGTM 👍

**Pull request merged**